include/deal.II/matrix_free/operators.h

```diff
@@ -1103,10 +1114,19 @@ namespace MatrixFreeOperators
 
 
   template <typename OperatorType>
+  template <typename VectorType>
   void
-  MGInterfaceOperator<OperatorType>::Tvmult (LinearAlgebra::distributed::Vector<typename MGInterfaceOperator<OperatorType>::value_type> &dst,
-                                             const LinearAlgebra::distributed::Vector<typename MGInterfaceOperator<OperatorType>::value_type> &src) const
+  MGInterfaceOperator<OperatorType>::Tvmult (VectorType &dst,
+                                             const VectorType &src) const
   {
+#ifdef DEAL_II_WITH_CXX11
+#ifndef DEAL_II_MSVC
```

Comment:
Since we need to work around the `static_assert` (and `CXX11`) at every place of `static_assert`, I definitely agree with this suggestion. Anyone else with an opinion on this?